src/server/utils/W3Helper.js

```diff
@@ -4,99 +4,114 @@ import conf from '../server.config'
 import logger from '../../imports/pino-logger'
 
 const log = logger.child({ from: 'W3Helper' })
+const Timeout = timeout => {
+  return new Promise((res, rej) => {
+    setTimeout(rej, timeout, new Error('Request Timeout'))
+  })
+}
 
-const w3PUTUserReq = (user, _options) => {
-  const options = {
-    getResponse: false,
-    ..._options
-  }
-  const secureHash = md5(user.email + conf.secure_key)
+export default {
+  baseUrl: `${conf.web3SiteUrl}/api/wl/user`,
+  baseHeaders: {
+    'Content-Type': 'application/json'
+  },
 
-  return new Promise((resolve, reject) => {
-    fetch(`${conf.web3SiteUrl}/api/wl/user`, {
-      method: 'PUT',
-      headers: {
-        'Content-Type': 'application/json'
-      },
-      body: JSON.stringify({
-        secure_hash: secureHash.toLowerCase(),
-        ...user
+  baseQuery(url, headers, body, method = 'post', options = {}, timeout = 15000) {
+    const fullUrl = `${this.baseUrl}${url}`
+    const stringBody = method !== 'GET' ? JSON.stringify(body) : undefined
+
+    log.debug('req options', { url, headers, body, method, options, timeout })
+
+    return Promise.race([Timeout(timeout), fetch(fullUrl, { method, body: stringBody, headers })])
+      .then(async res => {
+        log.debug('request response', res)
+
+        if (res.status >= 300) {
+          throw new Error(await res.text())
+        }
+
+        return res.json()
       })
-    })
-      .then(r => r.json())
       .then(response => {
         let toReturn = response.data
 
         if (options.getResponse) {
           toReturn = response
         }
 
-        resolve(toReturn)
+        return toReturn
       })
       .catch(e => {
-        log.error('Fetch W3 User Failed', e.message, e)
+        log.trace(e)
 
-        reject(e)
+        throw e
       })
-  })
-}
+  },
 
-export const registerUser = (user, options) => {
-  return w3PUTUserReq(
-    {
-      email: user.email,
-      full_name: user.fullName,
-      wallet_address: user.gdAddress
-    },
-    options
-  )
-}
+  w3PUTUserReq(user, _options) {
```

Comment:
we dont need this method

Reply:
@sirpy why? I made it to prevent code duplicating. The secure hash generating here which used for both registerUser and getLoginOrWalletToken methods.